kratos/solving_strategies/strategies/explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,412 @@
+
+        // Perform the intermidate sub step update
+        InitializeRungeKuttaIntermediateSubStep();
+        p_explicit_bs->BuildRHS(r_model_part);
+
+#pragma omp parallel for firstprivate(dof_size)
+        for (int i_dof = 0; i_dof < dof_size; ++i_dof) {
+            auto it_dof = r_dof_set.begin() + i_dof;
+            // Save current value in the corresponding vector
+            const double& r_res = it_dof->GetSolutionStepReactionValue();
+            rIntermediateStepResidualVector(i_dof) = r_res;
+            // Do the DOF update
+            double& r_u = it_dof->GetSolutionStepValue(0);
+            const double& r_u_old = it_dof->GetSolutionStepValue(1);
+            if (!it_dof->IsFixed()) {
+                const double mass = r_lumped_mass_vector(i_dof);
+                r_u = r_u_old + SubStepCoefficient * (dt / mass) * r_res;
+            } else {
+                const double delta_u = rFixedDofsValues(i_dof) - r_u_old;
+                r_u = r_u_old + SubStepCoefficient * delta_u;
+            }
+        }
+
+        FinalizeRungeKuttaIntermediateSubStep();
+    }
+
+    /**
+     * @brief Performs the last RK4 step
+     * This functions performs all the operations required in the last RK4 sub step
+     * @param rLastStepResidualVector The vector to store the last sub step residual
+     */
+    virtual void PerformRungeKuttaLastSubStep(LocalSystemVectorType& rLastStepResidualVector)
+    {
+        // Get the required data from the explicit builder and solver
+        const auto p_explicit_bs = BaseType::pGetExplicitBuilderAndSolver();
+        auto& r_dof_set = p_explicit_bs->GetDofSet();
+        const unsigned int dof_size = p_explicit_bs->GetEquationSystemSize();
+
+        // Get model part
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+
+        // Set the RUNGE_KUTTA_STEP value. This has to be done prior to the InitializeRungeKuttaStep()
+        r_process_info.GetValue(RUNGE_KUTTA_STEP) = 4;
+
+        // Perform the last sub step residual calculation
+        InitializeRungeKuttaLastSubStep();
+        p_explicit_bs->BuildRHS(r_model_part);
+
+#pragma omp parallel for firstprivate(dof_size)
+        for (int i_dof = 0; i_dof < dof_size; ++i_dof) {
+            const auto it_dof = r_dof_set.begin() + i_dof;
+            // Save current value in the corresponding vector
+            const double& r_res = it_dof->GetSolutionStepReactionValue();
+            rLastStepResidualVector(i_dof) = r_res;
+        }
+
+        FinalizeRungeKuttaLastSubStep();
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    const double mA21 = 0.5; // RK4 a_21 coefficient
```

Comment:
I have mixed feelings about these coefficients being const and member variable...

Reply:
You would make them optional to use other RK schemes?

Reply:
I would create a method to initialise the variables and call it in the Initialize or in the constructor. IMO this would give more flexibility. I am not an expert in RK schemes, but maybe it is possible to modify slightly the coefficients to get different types of schemes (e.g numerically damped)

Reply:
@loumalouomega : I think RK4 should not be modified, it is what it is. If you want to modify it, change its name or look for a more general scheme that has RK4 as a particular case.

Reply:
On top of Guillermo's comment, I'd like to add that other RK schemes require storing the intermediate residuals, something that becomes in a huge memory consumption. Having said this, I'd keep it as it is.